controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java

```diff
@@ -127,6 +130,12 @@ public boolean authorize(String resource, Principal principal, String credential
         return handler.authorize(resource, principal).ordinal() >= level.ordinal();
     }
 
+    @Synchronized
```

Comment:
No. `@Synchronized` will create a field named `$lock` and synchronize the method on that one.

If you look at the definition of `handlerMap`, it is `GuardedBy(this)`. So you either need to add the `synchronized` modifier (not annotation) to the method or use a `synchronized(this)` block inside it.

Doesn't your IDE highlight this problem? You can configure IntelliJ to highlight concurrency issues. You can find that in Settings->Editor.

Reply:
Thanks for catching it. Will fix it in the next commit. And, thanks for the suggestion about enabling it in IntelliJ - I didn't know it had something like that (mostly had been Eclipse and Visual Studio user).